### PR TITLE
Add a function to return 1.-filled response

### PR DESCRIPTION
**Pull request by jedori0228**
A function added in ISystProviderTool (`ISystProviderTool::GetDefaultEventResponse()`) that returns 1.-filled weight vector, where the number of elements corresponds to the number of SystParamHeader that are set.

### CODE REVIEW

File: systematicstools/interface/ISystProviderTool.hh [hunk 1]
```diff
@@ -115,6 +115,10 @@ public:
 #ifndef NO_ART
   virtual std::unique_ptr<EventResponse>
   GetEventResponse(art::Event const &) = 0;
+
+  //==== return 1-filled event_unit_response_t
+  systtools::event_unit_response_t GetDefaultEventResponse();
```
Comment (knoepfel) [inline systematicstools/interface/ISystProviderTool.hh]: Recommend making this `const`-qualified:

```suggestion
  systtools::event_unit_response_t GetDefaultEventResponse() const;
```

File: systematicstools/interface/ISystProviderTool.cc [hunk 2]
```diff
@@ -110,6 +110,28 @@ bool ISystProviderTool::ConfigureFromParameterHeaders(
...
+      resp.back().responses.push_back(1.);
+    } else {
+      for ( std::vector<double>::size_type i_var=0; i_var<sph.paramVariations.size(); i_var++) {
+        resp.back().responses.push_back(1.);
+      }
+    }
+
+  }
+  return resp;
+}
```
Comment (knoepfel) [inline systematicstools/interface/ISystProviderTool.cc]: This member function is more complicated than it needs to be.  I suggest a simpler implementation, broken into two functions:

```c++
ParamResponses 
responses_for(SystParamHeader const& sph)
{
  if (sph.isCorrection) {
    return {sph.systParamId, std::vector{1.}};
  }
  return {sph.systParamId, std::vector<double>(sph.paramVariations.size(), 1.)};
}

systtools::event_unit_response_t
ISystProviderTool::GetDefaultEventResponse() const 
{
  auto const& smd = this->GetSystMetaData();
  systtools::event_unit_response_t resp;
  resp.reserve(smd.size());
  for(auto const& sph : smd) {
    resp.push_back(response_for(sph));
  }
  return resp;
}
```

**Review by knoepfel — COMMENTED** [review]
Thanks for the PR, @jedori0228.  Please take a look at the comments below.

File: systematicstools/interface/ISystProviderTool.hh (hunk 1 above)
Comment (jedori0228) [reply systematicstools/interface/ISystProviderTool.hh]: `const` qualifiers are added; functions related to `GetDefaultEventResponse()` are also modified for this.

File: systematicstools/interface/ISystProviderTool.cc (hunk 2 above)
Comment (jedori0228) [reply systematicstools/interface/ISystProviderTool.cc]: Thank you! This is added in the new commit.

**Comment by knoepfel** [comment]
We will consider merging this pending the design discussion mentioned [here](https://github.com/LArSoft/nusystematics/pull/2#issuecomment-1154092389).

**Comment by lgarren** [comment]
approve

**Pull request merged**